ray-curator/ray_curator/utils/performance_utils.py

```diff
@@ -41,6 +56,10 @@ def __add__(self, other: StagePerfStats) -> StagePerfStats:
             actor_idle_time=self.actor_idle_time + other.actor_idle_time,
             input_data_size_mb=self.input_data_size_mb + other.input_data_size_mb,
             num_items_processed=self.num_items_processed + other.num_items_processed,
+            custom_stats={
```

Comment:
IMO, we should not add if the keys are not exactly the same. StagePerf is a task-level metrics and I think they should have the same keys.

Reply:
> IMO, we should not add if the keys are not exactly the same
They're only being added when keys are the same, so I'm confused.

Also it seems like __add__ is never used so I'm okay to get rid of it too